File: reservation-frontend/src/app/request-commands/editReservationCommand.ts
import { Guid } from "guid-typescript"
import { Ticket } from "../models/ticket"

export class EditReservationCommand {
    constructor(
        id: number, 
        name: string, 
        email: string, 
        phoneNumber: string, 
        ticketCount: number, 
        eventOccurrenceId: number, 
        paymentCompleted: boolean, 
        isDeleted: boolean,
        tickets: Ticket[],
        userId: Guid) {
        this.id = id
        this.name = name
        this.email = email
        this.phoneNumber = phoneNumber
        this.ticketCount = ticketCount
        this.eventOccurrenceId = eventOccurrenceId
        this.paymentCompleted = paymentCompleted
        this.isDeleted = isDeleted
        this.tickets = tickets
        this.userId = userId
    }
    public id!: number
    public name!: string
    public email!: string
    public phoneNumber!: string
    public ticketCount!: number
    public eventOccurrenceId!: number
    public paymentCompleted!: boolean
    public isDeleted!: boolean
    public tickets!: Ticket[]
    public userId!: Guid
}

File: reservations-api/src/Reservations.Application/Reservations/Commands/EditReservation/EditReservationCommand.cs
using MediatR;
using Reservations.Application.DataTransferObjects;
using Reservations.Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reservations.Application.Reservations.Commands.EditReservation
{
    public class EditReservationCommand : IRequest
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public IEnumerable<Ticket> Tickets { get; set; }
        public bool PaymentCompleted { get; set; }
        public long EventOccurrenceId { get; set; }
        public bool IsDeleted { get; set; }
        public Guid UserId { get; set; }
    }
}
